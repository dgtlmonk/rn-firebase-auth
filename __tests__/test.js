import React from "react";
import { Button, Text, TextInput, View } from "react-native";
import { fireEvent, render, wait } from "@testing-library/react-native";

function Example() {
  const [name, setUser] = React.useState("");
  const [show, setShow] = React.useState(false);

  return (
    <View>
      <TextInput value={name} onChangeText={setUser} testID="input" />
      <Button
        title="Print Username"
        onPress={() => {
          // let's pretend this is making a server request, so it's async
          // (you'd want to mock this imaginary request in your unit tests)...
          setTimeout(() => {
            setShow(!show);
          }, Math.floor(Math.random() * 200));
        }}
      />
      {show && <Text testID="printed-username">{name}</Text>}
    </View>
  );
}

test("examples of some things", async () => {
  const { getByTestId, getByText, queryByTestId, baseElement } = render(
    <Example />
  );
  const famousWomanInHistory = "Ada Lovelace";

  const input = getByTestId("input");
  fireEvent.changeText(input, famousWomanInHistory);

  const button = getByText("Print Username");
  fireEvent.press(button);

  await wait(() => expect(queryByTestId("printed-username")).toBeTruthy());

  expect(getByTestId("printed-username").props.children).toBe(
    famousWomanInHistory
  );
  expect(baseElement).toMatchSnapshot();
});
