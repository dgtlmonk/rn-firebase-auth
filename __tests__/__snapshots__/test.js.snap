// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`examples of some things 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View>
      <TextInput
        allowFontScaling={true}
        rejectResponderTermination={true}
        testID="input"
        underlineColorAndroid="transparent"
        value="Ada Lovelace"
      />
      <Button
        title="Print Username"
      />
      <Text
        testID="printed-username"
      >
        Ada Lovelace
      </Text>
    </View>
  </View>
</View>
`;
